app/src/main/java/com/m2049r/xmrwallet/util/LocaleHelper.java

```diff
@@ -0,0 +1,85 @@
+package com.m2049r.xmrwallet.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.res.Configuration;
+import android.preference.PreferenceManager;
+
+import com.m2049r.xmrwallet.R;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Locale;
+
+public class LocaleHelper {
+    private static final String PREFERRED_LOCALE_KEY = "preferred_locale";
+    private static Locale SYSTEM_DEFAULT_LOCALE = Locale.getDefault();
+
+    public static ArrayList<Locale> getAvailableLocales(Context context) {
+        ArrayList<Locale> locales = new ArrayList<>();
+        HashSet<String> localizedStrings = new HashSet<>();
+        int compareResId = R.string.menu_about;
```

Comment:
is this really the correct string to compare? i am guessing some languages have the same string for "About" - maybe we should add a specialized string to id the translations, like ```R.string.language``` which can maybe also be used for the display in the dialog list ?

Reply:
It's a good idea.

If there is no better idea and you are sure to do this, i will modify code and add resource like ```<string name="language">English</string>``` to all strings.xml